example/test_driver/cloud_firestore_behaviors.dart

```diff
@@ -126,6 +127,31 @@ void main() {
       expect(timeDiff.inMilliseconds, _test.lessThanOrEqualTo(1));
     });
 
+    void setupFieldValueFactoryForFirestore(Firestore firestore) {
+      if (firestore is MockFirestoreInstance) {
+        mockFirestore.setupMockFieldValueFactory();
+      } else {
+        mockFirestore.restoreFieldValueFactory();
+      }
+    }
```

Comment:
@atn832 Do you know come up with a good way to test FieldValue in this integration tests with the 3 Firestore instances? As of now (master, 20b86be8dc144606c63d2d703d46d3275dd3d15f), this integration test cannot run with FieldValue operations. This is because `MockFirestoreInstance._setupFieldValueFactory` prevents real Firestore from using real FieldValueFactory (`MethodChannelFieldValueFactory`).

I thought this commit (fe04462ded29f090e5c332ff7df3b3d7f3da65b1) would enable all three Firebase instances with their FieldValues. However, this cannot update `FieldValue`'s `static final _factory` field.

```
class FieldValue extends platform.FieldValuePlatform {
  static final platform.FieldValueFactoryPlatform _factory =
      platform.FieldValueFactoryPlatform.instance;
```

Error message: https://github.com/atn832/cloud_firestore_mocks/pull/50/commits/c00c28db13a9733e5cba685f2c062ce512281de5

Do you think of a good idea to run this integration test with FieldValue?

Reply:
Idea: for FieldValue, separate invocations between real Firestore and the mock firestore, while using the same set of test cases.

Reply:
Yes instead of running integration tests once with a loop for each implementation, you could run integration tests 3 times, each with a different instance. Is it possible to pass parameters from the command line when running integration tests?